app/models/reporting/weekly_report.rb

```diff
@@ -0,0 +1,57 @@
+module Reporting
+  class WeeklyReport
+    include Reporting::Reportable
+
+    PARAM_FORMAT = '%G-%V'.freeze
+
+    def initialize(date:, report_type:)
+      @date = date
+      @report_type = Types::WeeklyReportType[report_type]
+    end
+
+    attr_reader :date, :report_type
+
+    def to_param
+      date.strftime(PARAM_FORMAT)
+    end
+
+    def week_name
+      date.strftime('Week %V, %G')
+    end
+
+    def title
+      report_text(:title, week_name:)
+    end
+
+    def start_on
+      date.beginning_of_week
+    end
+
+    def end_on
+      date.end_of_week
+    end
+
+    private
+
+    def stream_name
+      return nil unless report_type == 'caseworker_report'
```

Comment:
Could we avoid hardcoding `caseworker_report`

Reply:
I've removed this guard as it is no longer needed.